lib/api/bucketPut.js

```diff
@@ -97,6 +97,49 @@ function _parseXML(request, log, cb) {
     });
 }
 
+function _buildConstantParams({ request, bucketName, authInfo, authParams, ip, locationConstraint, apiMethod }) {
+    return {
+        constantParams: {
+            headers: request.headers,
+            query: request.query,
+            generalResource: bucketName,
+            specificResource: {
+                key: '',
+            },
+            requesterIp: ip,
+            sslEnabled: request.connection.encrypted,
+            awsService: 's3',
+            requesterInfo: authInfo,
+            signatureVersion: authParams.params.data.authType,
+            authType: authParams.params.data.signatureVersion,
+            signatureAge: authParams.params.data.signatureAge,
+            apiMethod,
+            locationConstraint,
+        },
+    };
+}
+
+function _handleAuthResults(locationConstraint, log, next) {
+    return (err, authorizationResults) => {
+        if (err) {
+            return next(err);
+        }
+        if (authorizationResults[0].isAllowed !== true) {
```

Comment:
I would feel more confident if we can unit test this, or add a check to ensure `authorizationResults` is an array, to avoid errors.

Also, if we have more than one result, shouldn't we iterate over them? If we have one of the permissions not set, we should deny the overall action, right?

Reply:
Check policies in vault has always returned an array, but I swapped out the [0] with a .every and added a few unit tests.